examples/embeddable_examples/public/app/presentation_container_example/components/top_nav.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { useEffect, useState } from 'react';
+import useMountedState from 'react-use/lib/useMountedState';
+import { EuiBadge, EuiButton, EuiButtonEmpty, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { PublishesUnsavedChanges } from '@kbn/presentation-publishing';
+
+interface Props {
+  onSave: () => Promise<void>;
+  resetUnsavedChanges: () => void;
+  unsavedChanges$: PublishesUnsavedChanges['unsavedChanges'];
+}
+
+export function TopNav(props: Props) {
+  const isMounted = useMountedState();
+  const [isSaving, setIsSaving] = useState(false);
+  const [hasUnsavedChanges, setHasUnsavedChanges] = useState(false);
+  useEffect(() => {
+    const subscription = props.unsavedChanges$.subscribe((unsavedChanges) => {
+      setHasUnsavedChanges(unsavedChanges !== undefined);
+    });
```

Comment:
This is still reporting unsaved changes when resetting after adding new panels


https://github.com/user-attachments/assets/70a04768-b8d7-4162-9e25-bfbeec16528a

Reply:
resolved with eb6d27c480aadb8722efe4801f4cc67b491a5193